src/librustc_mir/borrow_check/diagnostics/mod.rs

```diff
@@ -566,31 +566,52 @@ pub(super) enum UseSpans {
 }
 
 #[derive(Copy, Clone, PartialEq, Eq, Debug)]
-pub(super) enum FnSelfUseKind {
+pub(super) enum FnSelfUseKind<'tcx> {
     /// A normal method call of the form `receiver.foo(a, b, c)`
-    Normal { self_arg: Ident, implicit_into_iter: bool },
+    Normal {
+        self_arg: Ident,
+        implicit_into_iter: bool,
+    },
     /// A call to `FnOnce::call_once`, desugared from `my_closure(a, b, c)`
     FnOnceCall,
     /// A call to an operator trait, desuraged from operator syntax (e.g. `a << b`)
-    Operator { self_arg: Ident },
+    Operator {
+        self_arg: Ident,
+    },
+    DerefCoercion {
+        deref_target: Span,
+        deref_ty: Ty<'tcx>,
```

Comment:
```suggestion
        deref_origin_ty: Ty<'tcx>,
```
or something like this, it wasn't instantly clear to me if this is `T` or `T::Deref`

Reply:
This is `T::Deref`, so I'll rename it to `deref_target_ty`.